SprawdzanieNIPu/ConsoleApplication1/NIP.cs

```diff
@@ -1,72 +1,68 @@
-﻿using System;
+﻿namespace ConsoleApplication1
+ {
+    using System;
 
-namespace ConsoleApplication1
-{
-    class NIP
-    {
-        public string numer;
-
-        public NIP(string nr)
-        {
-            this.numer = nr.Trim();
-        }
-
-        public Boolean dlugoscOK
-        {
-            get
-            {
-                return this.numer.Length == 10;
-            }
-        }
-
-        public int kontrolna
-        {
-            // zwraca sumę kontrolną
-            get
-            {
-                int[] wagi = new int[] {6, 5, 7, 2, 3, 4, 5, 6, 7};
-                double x=0;
-                double liczba=0;
+ /// <summary>
+ /// Klasa służy do sprawdzenia, czy wprowadzony ciąg znaków jest poprawnym numerem NIP i jego sformatowania.
+ /// </summary>
+ internal class NIP
+ {
+   private long numer;
+   private int[] wagi = new int[] { 6, 5, 7, 2, 3, 4, 5, 6, 7 };
+   private int kontrolna;
 
-                for (int i = 0; i < wagi.Length; i++) 
-                {
-                    x = Convert.ToDouble(numer.Substring(i, 1));
-                    liczba = liczba + x*wagi[i];
-                }
+   private bool czyPoprawny;
+   private string sformatowana;
 
-                liczba = liczba % 11;
+   public bool CzyPoprawny
+   {
+     get 
+     {
+        return this.czyPoprawny;
+     }
+    }
 
-                return (int)liczba;
-            }
+    public string Sformatowana
+    {
+      get 
+       {
+         return this.sformatowana;
         }
-
-        public Boolean czyPoprawna
-        {
-            get
+     }
+            
+   public NIP(string nr)
+   {
+      nr = nr.Trim();
+            long x;
+            if (nr.Length == 10 && long.TryParse(nr.Trim(), out x))
             {
-               return this.dlugoscOK && 
-                   (this.kontrolna == Convert.ToDouble(this.numer.Substring(9, 1)));
+                this.numer = x;
+                this.sformatowana = string.Format("{0:000-000-00-00}", this.numer);
+                this.kontrolna = this.ObliczKontrolna(this.numer);
+                this.czyPoprawny = this.kontrolna == (int)NtaCyfra(this.numer, 9);
             }
-        }
+      else
+      {
+         this.sformatowana = nr;
```

Comment:
Nie podoba mi się ten pomysł - poprawne to formatuję, niepoprawny to nie formatuję.